charts/seed-bootstrap/charts/monitoring/templates/config.yaml

```diff
@@ -68,6 +68,28 @@ data:
         - prometheus-web.garden.svc
       metric_relabel_configs:
 {{ include "prometheus.keep-metrics.metric-relabel-config" .Values.allowedMetrics.cadvisor | indent 6 }}
+
+    - job_name: kube-state-metrics
+      # Service is used, because we only care about metric from one kube-state-metrics instance
+      # and not multiple in HA setup
+      kubernetes_sd_configs:
+      - role: service
+        namespaces:
+          names: [ garden ]
+      relabel_configs:
+      - source_labels: [ __meta_kubernetes_service_label_component ]
+        regex: kube-state-metrics
+        action: keep
+      - source_labels: [ __meta_kubernetes_service_port_name ]
+        regex: metrics
+        action: keep
+      - target_label: instance
+        replacement: kube-state-metrics
+      metric_relabel_configs:
+      - source_labels: [ namespace ]
+        regex: shoot-.+
```

Comment:
Why did you remove the start/end tokens (`^...$`)? Does this match `extension-extension-shoot-dns-service-abcdef` now?

Reply:
Prometheus automatically anchors the regex. So it will not match the namespace that you mentioned. https://prometheus.io/docs/prometheus/latest/configuration/configuration/#regex

Reply:
Ok, perfect! 👍